invenio_communities/models.py

```diff
@@ -311,11 +316,19 @@ def remove_record(self, record):
         :type record: `invenio_records.api.Record`
         """
         key = current_app.config['COMMUNITIES_RECORD_KEY']
+        access_key = '_access'
 
         assert self.id in record.get(key, [])
-
         record[key] = [c for c in record[key] if c != self.id]
 
+        assert self.id in record.get(access_key, {'read': []})['read']
+        record[access_key]['read'] = \
+            [c for c in record[access_key]['read'] if c != self.id]
```

Comment:
then you can just use the first solution I give which may be more time consuming but we don't really care here, but is very clear when you read the code ;)